src/main/java/Task.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Class that encapsulates a task that can be added to `Torne`.
+ */
+public class Task {
+    protected final String name;
+    protected boolean isDone;
+
+    /**
+     * Creates a Task object from a given storage string.
+     *
+     * @return Task created from the input string.
+     * @throws TorneInvalidDataException if the input string is invalid.
+     */
+    public static Task fromStorageString(String input) throws TorneInvalidDataException, TorneInvalidCommandException {
+        String[] parts = input.split(" / ");
+        if (parts.length < 3) {
+            // min length is 3. If less than that, raise error
+            throw new TorneInvalidDataException("Input task string does not have enough components");
+        }
+
+        boolean isTaskDone;
```

Comment:
Nice naming of the boolean isDone as well as the Exceptions. The naming of the InvalidData and InvalidCommand Exceptions are clear to understand